internal/smartlink/smartlink.go

```diff
@@ -39,13 +39,26 @@ func LinkContents(src, dest string) error {
 
 // Link creates a link from src to target. MS decided to support Symlinks but only if you opt into developer mode (go figure),
 // which we cannot reasonably force on our users. So on Windows we will instead create dirs and hardlinks.
-func Link(src, dest string) error {
+func Link(src, dest string, visited map[string]bool) error {
+	srcWasSymlink := isSymlink(src)
+
 	var err error
 	src, dest, err = resolvePaths(src, dest)
 	if err != nil {
 		return errs.Wrap(err, "Could not resolve src and dest paths")
 	}
 
+	if visited == nil {
+		visited = make(map[string]bool)
+	}
+	if _, exists := visited[src]; exists && srcWasSymlink {
```

Comment:
Hmm I think this is now trying to solve the same issue two ways. If the src is a symlink then we should just make the deployment a link also, and stop iterating further down because it's a symlink. In this case what we already visited is irrelevant.

So I think we can get rid of the visited logic altogether, and simply revise it to make dir links if the src is a dir link, and otherwise create the dir and continue recursion.